lib/PnpPlugin.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const path = require("path");
+
+module.exports = class PnpPlugin {
+	constructor(source, pnpApi, target) {
+		this.source = source;
+		this.pnpApi = pnpApi;
+		this.target = target;
+	}
+
+	apply(resolver) {
+		const target = resolver.ensureHook(this.target);
+		resolver
+			.getHook(this.source)
+			.tapAsync("PnpPlugin", (requestContext, resolveContext, callback) => {
+				let request = requestContext.request;
+				let issuer = requestContext.context.issuer;
```

Comment:
`issuer` is something webpack related. enhanced-resolve doesn't know about it (and should not rely on it).

Please always use `requestContext.path`.